tests/pyfunc/sample_code/func_code.py

```diff
@@ -1,8 +1,8 @@
 from mlflow.models import set_model
 
 
-def predict(model_input):
-    return f"This was the input: {model_input}"
+def predict(model_input: list[str]):
```

Comment:
Could we test both? with and without type hint

Reply:
When the type hint is omitted, the function will receive the Pandas series, do we want to test that flow?

Reply:
Yes since it's what it is today. Let's also make sure model signature is inferred from input example if no type hint is provided

Reply:
Added another usecase in d7cbc9629a0f0e65dd78304983fde983b9cb3c17